File: 0881 boats-to-save-people/solution.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people)
    i,j,c:=0,len(people)-1,0
    for i<=j {
        if people[i]+people[j]<=limit {
            i++
        }
        j--
        c++
    }
    return c
}

File: 0881 boats-to-save-people/solution.ts
function numRescueBoats(p: number[], li: number): number {
    p.sort((a,b)=>a-b);
    let l=0,r=p.length-1,c=0;
    while(l<=r) {
        let sum = p[l] + p[r];
        if(sum <= li) l++;
        r--;
        c++; 
    }
    return c;
};
